front-end/src/components/newAccountForm.tsx

```diff
@@ -0,0 +1,188 @@
+    console.log(`Email: ${emailField}`);
+    console.log(`Password: ${passwordField}`);
+    console.log(`Check Password: ${passwordField2}`);
+
+    //All fields must be filled
+    if (firstNameField  === "" || lastNameField  === "" || emailField === "" || passwordField === "" || passwordField2 === "") {
+      setAlertReason("Enter your name, email address, and password.");
+      return;
+    }
+
+    //Check username/password requirements and security for text fields.
+    //Firstname and Lastname forms must use appropriate signage (alpha characters only)
+    if (!(firstNameField.match(checkSignage)) || !(lastNameField.match(checkSignage)) ) {
+      setAlertReason("Name fields must only be alphabetic characters.");
+      return;
+    }
+
+    //Email should only allow basic signage.
+    if (!(emailField.match(checkEmailSignage)) ) {
+      setAlertReason("Please enter a valid email address.");
+      return;
+    }
+
+    //Passwords must match
+    if (passwordField !== passwordField2) {
+      setAlertReason("Your passwords must match.");
+      return;
+    }
+
+    //Passwords must be correct format (8-20 characters, 1 letter, 1 number, 1 special character, no spaces)
+    if (!(passwordField.match(checkPassSignage)) || !(passwordField2.match(checkPassSignage)) ) {
+      setAlertReason("Passwords must be 6-20 characters, and contain 1 letter, 1 number, 1 special character (@$!%*#?&^), and no spaces.");
+      return;
+    }
+
+    //console.log info should be passed to the backend here.
+
+    history.push("home");
+  };
+
+  const handleAlertClose = (
+    event?: React.SyntheticEvent | Event,
+    reason?: string
+  ) => {
+    if (reason === "clickaway") {
+      return;
+    }
+
+    setAlertReason("");
+  };
+
+  return (
+    <FormGroup style={styles.signIn}>
+      <TextField
+        onChange={handleFirstNameChange}
+        style={styles.input}
+        id="firstName"
+        label="First Name"
+        variant="outlined"
+      />
+      <TextField
+        onChange={handleLastNameChange}
+        style={styles.input}
+        id="lastName"
+        label="Last Name"
+        variant="outlined"
+      />
+      <TextField
+        onChange={handleEmailChange}
+        style={styles.input}
+        id="email"
+        label="Email"
+        variant="outlined"
+      />
+      <TextField
+        onChange={handlePasswordChange}
+        style={styles.input}
+        id="password"
+        label="Password"
+        variant="outlined"
+        type="password"
+      />
+      <TextField
+        onChange={handlePassword2Change}
+        style={styles.input}
+        id="checkpassword"
+        label="Confirm Password"
+        variant="outlined"
+        type="password"
+      />
+      <div>
+        <Button
+          onClick={handleSubmit}
+          style={styles.button}
+          variant="contained"
+        >
+          Create Account
+        </Button>
+      </div>
+      <Link href="/" underline="always">
```

Comment:
Could we add a margin-up to the link? In this form and other one.

Reply:
Could you elaborate what you mean? I'm not sure what "a margin-up" is supposed to be, sorry.

Reply:
Resolved the issue by adding a new margin into the field.